float8_experimental/float8_ops.py

```diff
@@ -140,3 +141,29 @@ def autocast_to_copy(aten_op, args, kwargs=None):
     return Float8Tensor(
         args[0]._data, args[0]._scale, kwargs["dtype"], args[0]._emulate
     )
+
+
+@implements([c10d_functional.all_gather_into_tensor.default])
+def allgather_fp8(aten_op, args, kwargs=None):
+    """
+    override funcol with FP8 handling
+    """
+    fp8_input = args[0]
+    assert isinstance(fp8_input, Float8Tensor)
+
+    fp8_data = fp8_input._data
+    fp8_data = fp8_data.view(torch.int8)
+    fp8_data = fp8_data.contiguous()
```

Comment:
should this be an assert is.contiguous .. I guess not I think the _make_wrapper_subclass consumes the input data strides so should be fine